pkg/option/config.go

```diff
@@ -1654,6 +1671,17 @@ func (c *DaemonConfig) Populate() {
 		log.WithError(err).Fatalf("Unable to parse excluded local addresses")
 	}
 
+	c.IdentityAllocationMode = viper.GetString(IdentityAllocationMode)
+	switch {
+	// This is here for tests. Some call Populate without the normal init
+	case c.IdentityAllocationMode == "":
+		c.IdentityAllocationMode = IdentityAllocationModeKVstore
+
+	case c.IdentityAllocationMode != IdentityAllocationModeKVstore &&
```

Comment:
Nit: why not have, 
```
case IdentityAllocationModeKVStore, caseIdentityAllocationModeCRD:
default:
    log.Fatal(
...
```

It would be easier to parse / more in line with how we handle invalid config in other parts of the code.